src/shared/scratchOrgRequest.ts

```diff
@@ -22,7 +24,8 @@ export const buildScratchOrgRequest = async (
     ...(flags['definition-file']
       ? (JSON.parse(await fs.promises.readFile(flags['definition-file'], 'utf-8')) as Record<string, unknown>)
       : {}),
-    ...(flags.edition ? { edition: flags.edition } : {}),
+    ...(flags.edition && !flags.snapshot ? { edition: flags.edition } : {}),
```

Comment:
`&& !flags.snapshot` seems unnecessary since oclif won't allow this case per the flag definition.